core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -57,7 +57,10 @@ class AdminClient(val time: Time,
       now = time.milliseconds()
     } while (now < deadline && future.exception().isInstanceOf[SendFailedException])
 
-    throw future.exception()
+    if(future.failed())
```

Comment:
Nitpick: there should be a space after the `if` (I know there are cases where our codebase is not consistent with regards to this, but we're trying to improve).